Scripts/models/logit.py

```diff
@@ -0,0 +1,155 @@
+            try:
+                exps *= numpy.power(impedance[i] + 1, b[i])
+            except ValueError:
+                exps[:k, :] *= numpy.power(impedance[i][:k, :] + 1, b[i][0])
+                exps[k:, :] *= numpy.power(impedance[i][k:, :] + 1, b[i][1])
+        return exps
+    
+    def add_zone_util(self, utility, b, generation=False):
+        zdata = self.zone_data
+        k = zdata.zone_numbers.get_loc(parameters.first_surrounding_zone)
+        for i in b:
+            try:
+                utility += b[i] * zdata.get_data(i, self.purpose, generation)
+            except ValueError:
+                data_capital_region = zdata.get_data(i, self.purpose, generation, 0)
+                data_surrounding = zdata.get_data(i, self.purpose, generation, 1)
+                if utility.ndim == 1:
+                    utility[:k] += b[i][0] * data_capital_region
+                    utility[k:] += b[i][1] * data_surrounding
+                else:
+                    utility[:k, :] += b[i][0] * data_capital_region
+                    utility[k:, :] += b[i][1] * data_surrounding
+        return utility
+
+
+class ModeDestModel(LogitModel):
+    def calc_prob(self, impedance):
+        dest_expsums = {}
+        for mode in self.dest_choice_param:
+            expsum = self.calc_dest_util(mode, impedance[mode])
+            dest_expsums[mode] = {}
+            dest_expsums[mode]["logsum"] = expsum
+        mode_expsum = self.calc_mode_util(dest_expsums)
+        prob = {}
+        for mode in self.mode_choice_param:
+            mode_prob = self.mode_exps[mode] / mode_expsum
+            dest_expsum = dest_expsums[mode]["logsum"]
+            dest_prob = self.dest_exps[mode].T / dest_expsum
+            prob[mode] = mode_prob * dest_prob
+        return prob
+
+
+class DestModeModel(LogitModel):
+    def calc_prob(self, impedance):
+        mode_expsum = self.calc_mode_util(impedance)
+        logsum = {"logsum": mode_expsum}
+        dest_expsum = self.calc_dest_util("logsum", logsum)
+        prob = {}
+        dest_prob = self.dest_exps["logsum"].T / dest_expsum
+        for mode in self.mode_choice_param:
+            mode_prob = (self.mode_exps[mode] / mode_expsum).T
+            prob[mode] = mode_prob * dest_prob
+        return prob
+
+class SecDestModel(LogitModel):
+    def calc_prob(self, mode, impedance):
+        expsum = self.calc_dest_util(mode, impedance)
+        prob = self.dest_exps[mode].T / expsum
+        return prob
+
+class OriginModel(LogitModel):
+    def calc_prob(self, impedance):
+        b = self.dest_choice_param
+        utility = self.calc_origin_util(impedance)
+        exps = numpy.exp(utility)
+        # Here, size means kokotekija in Finnish
+        size = numpy.ones_like(exps)
+        size = self.add_zone_util(size, b["size"])
+        exps *= numpy.power(size, b["log"]["size"])
+        expsums = numpy.sum(exps, axis=0)
+        prob = {}
+        # Mode is needed here to get through tests even
```

Comment:
If the mode is indeed unnecessary, I'd fix the tests according to current implementation.